cmake/CompilerWarnings.cmake

```diff
@@ -0,0 +1,80 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2013-2022 The Foundry Visionmongers Ltd
+
+function(set_default_compiler_warnings target_name)
+    option(WARNINGS_AS_ERRORS "Treat compiler warnings as errors" ON)
+
+    set(MSVC_WARNINGS
```

Comment:
[fix] Can we resolve this by simply removing the `MSVC` warning config until we actually start compiling on win?

Reply:
Removed the Windows warning config, leaving a warning message for when we come to compile on Windows, in 87bdfed